source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/imu/imu.py

```diff
@@ -0,0 +1,209 @@
+        if not hasattr(self, "_dt"):
+            raise RuntimeError(
+                "The update function must be called before the data buffers are accessed the first time."
+            )
+        # obtain the poses of the sensors
+        pos_w, quat_w = self._view.get_transforms()[env_ids].split([3, 4], dim=-1)
+        quat_w = math_utils.convert_quat(quat_w, to="wxyz")
+        # store the poses
+        self._data.pos_w[env_ids] = pos_w + math_utils.quat_rotate(quat_w, self._offset_pos_b)
+        self._data.quat_w[env_ids] = math_utils.quat_mul(quat_w, self._offset_quat_b)
+
+        # obtain the velocities of the sensors
+        lin_vel_w, ang_vel_w = self._view.get_velocities()[env_ids].split([3, 3], dim=-1)
+        # if an offset is present, the linear velocity has to be transformed taking the angular velocity into account
+        lin_vel_w += torch.cross(ang_vel_w, math_utils.quat_rotate(quat_w, self._offset_pos_b), dim=-1)
+        # obtain the acceleration of the sensors
+        lin_acc_w, ang_acc_w = self._view.get_accelerations()[env_ids].split([3, 3], dim=-1)
+        # if an offset is present, the linear acceleration has to be transformed taking the angular velocity and acceleration into account
+        lin_acc_w += torch.cross(ang_acc_w, math_utils.quat_rotate(quat_w, self._offset_pos_b), dim=-1) + torch.cross(
+            ang_vel_w, torch.cross(ang_vel_w, math_utils.quat_rotate(quat_w, self._offset_pos_b), dim=-1), dim=-1
+        )
+        # store the velocities
+        self._data.lin_vel_b[env_ids] = math_utils.quat_rotate_inverse(self._data.quat_w[env_ids], lin_vel_w)
+        self._data.ang_vel_b[env_ids] = math_utils.quat_rotate_inverse(self._data.quat_w[env_ids], ang_vel_w)
+        # store the accelerations
+        self._data.lin_acc_b[env_ids] = math_utils.quat_rotate_inverse(self._data.quat_w[env_ids], lin_acc_w)
+        self._data.ang_acc_b[env_ids] = math_utils.quat_rotate_inverse(self._data.quat_w[env_ids], ang_acc_w)
+
+    def _initialize_buffers_impl(self):
+        """Create buffers for storing data."""
+        # data buffers
+        self._data.pos_w = torch.zeros(self._view.count, 3, device=self._device)
+        self._data.quat_w = torch.zeros(self._view.count, 4, device=self._device)
+        self._data.quat_w[:, 0] = 1.0
+        self._data.lin_vel_b = torch.zeros(self._view.count, 3, device=self._device)
+        self._data.ang_vel_b = torch.zeros(self._view.count, 3, device=self._device)
+        self._data.lin_acc_b = torch.zeros(self._view.count, 3, device=self._device)
+        self._data.ang_acc_b = torch.zeros(self._view.count, 3, device=self._device)
+        # store sensor offset transformation
+        self._offset_pos_b = torch.tensor(list(self.cfg.offset.pos), device=self._device).repeat(self._view.count, 1)
+        self._offset_quat_b = torch.tensor(list(self.cfg.offset.rot), device=self._device).repeat(self._view.count, 1)
+
```

Comment:
```suggestion
# store COM offset
        com_pos_b, _ = self._view.get_coms().split([3, 4], dim=-1) # pull from cpu and then store in chosen device
        self._com_pos_b = torch.tensor(com_pos_b.tolist(),device=self._device).repeat(self._view.count, 1)
```

Reply:
that is not necessary for me, it captures the amount of views

Reply:
hmm interesting I had issues with get_coms() not excepting a cuda bound env_ids when calling `self._view.get_coms()[env_ids].split([3,4],dim=-1)`. It seemed to only work on cpu unless I did something like this. (this could be due to my naivete with torch device stuff though)

Reply:
for me it also returns the tensors on the CPU but a simple to(self.device) fixed it